cpp/src/arrow/util/sse_util.h

```diff
@@ -32,8 +32,15 @@
 #include <nmmintrin.h>
 #endif
 
+// Enable the SIMD for ByteStreamSplit Encoder/Decoder
+#define ARROW_HAVE_SIMD_SPLIT
```

Comment:
this seems superfluous, I think we can keep using the ARROW_HAVE_SSE4_2 where this is used, since that is the lowest level instruction set that is supported?  If you feel it is useful, I think this should be moved to the the new header file you introduced instead of the generic utilities.